File: exercises/D209/solution.pas
var
    i, n, t1, t2: longint;
    ok: boolean;
begin
    ok := true;
    readln(n);
    read(t1);
    if (t1 <= 100) or (t1 >= 50000) then
        ok := false;
    for i := 2 to n do
    begin
        read(t2);
        if (t2 <= 100) or (t2 >= 50000) or (t1 >= t2) then 
            ok := false;
        t1 := t2;
    end;
    if ok then 
        writeln('Valid') 
    else 
        writeln('Invalid')
end.

File: exercises/D209/solution.py
n = int(input())
nums = list(map(int, input().split()))
ok = True
if nums[0] <= 100 or nums[0] >= 50000:
    ok = False
prev = nums[0]
for i in range(1, n):
    if nums[i] <= 100 or nums[i] >= 50000 or prev >= nums[i]:
        ok = False
    prev = nums[i]
if ok:
    print('Valid')
else:
    print('Invalid')
